### PR TITLE
[WIP] Break the hypervisor dependency from the endpoint interface

### PR DESCRIPTION
**Pull request by sameo**
Fixes: #1209

### CODE REVIEW

File: virtcontainers/hypervisor.go [hunk 1]
```diff
@@ -552,9 +552,9 @@ type hypervisor interface {
 	pauseSandbox() error
 	saveSandbox() error
 	resumeSandbox() error
-	addDevice(devInfo interface{}, devType types.DeviceType) error
-	hotplugAddDevice(devInfo interface{}, devType types.DeviceType) (interface{}, error)
-	hotplugRemoveDevice(devInfo interface{}, devType types.DeviceType) (interface{}, error)
+	addDevice(types.Device) error
```
Comment (sboeuf) [inline virtcontainers/hypervisor.go]: Can you elaborate how this change will help decouple the hypervisor from the endpoint dependency?
From what I understand here, we're simply moving the device info as part of the device structure itself, but the detection of the device types such as `Endpoint` is still happening inside each hypervisor implementation, right?

**Comment by jodh-intel** [comment]
Ping @kata-containers/runtime - ...!

**Comment by jodh-intel** [comment]
@sameo - one file conflict to resolve.

**Comment by amshinde** [comment]
The Attach() for an endpoint stills invokes the hypervisor. How does this break the dependency on the hypervisor package?

**Comment by raravena80** [comment]
@sameo any updates? Ping. Thx!

**Comment by raravena80** [comment]
@sameo nudge

**Comment by raravena80** [comment]
@sameo any updates? Thx.

**Comment by raravena80** [comment]
Any takers? 😄

**Comment by raravena80** [comment]
Revisiting, any takers? 😄

**Comment by raravena80** [comment]
Any takers?

Your weekly Kata herder.

**Comment by raravena80** [comment]
Ping for any takers 😄

**Comment by raravena80** [comment]
Ping 😄 

Kata Herder.

**Comment by raravena80** [comment]
@jodh-intel this is old, should we close this and move the work to 2.0?

**Comment by fidencio** [comment]
As this PR is marked as do-not-merge, we're assuming it won't make for the final 1.13.0 release, thus we're closing it.
In case you feel this has been wrongly closed, please, feel free to re-open it.

**Pull request closed without merging**